lib/new-syntax/parser.lr

```diff
@@ -920,11 +920,27 @@ toplevel_chain_expression : Ast.ChainExpression = {
 
 chain_expression_list : Ast.Expression[] = {
     expr:projection_expression => [expr];
+    expr:join_expression => [expr];
 
     list:chain_expression_list '=>' expr:projection_expression => {
         list.push(expr);
         return list;
     };
+
+    list:chain_expression_list '=>' expr:join_expression => {
+        list.push(expr);
+        return list;
+    };
+}
+
+join_expression : Ast.Expression = {
+    lhs:projection_expression 'join' rhs:projection_expression => {
+        return new Ast.JoinExpression($.location, lhs, rhs, null, null);
+    };
+
+    lhs:projection_expression 'join' rhs:projection_expression 'on' condition:or_filter => {
+        return new Ast.JoinExpression($.location, lhs, rhs, condition, null);
+    };
```

Comment:
The way you wrote the grammar, `A join B join C` is not allowed, but the way you wrote toSource(), `(A join B) join C` will turn into `A join B join C` without parenthesis.

I think the right way to write the grammar is:
```
chain_expression_list = {
   join_expression;
   chain_expression_list => join_expression;
}
join_expression = {
   projection_expression;
   join_expression 'join' projection_expression;
   join_expression 'join' projection_expression 'on' or_filter;
}
```

Reply:
I actually changed the addParentheses function, so it will keep the parentheses in `(A join B) join C`. but your design is better

Reply:
hmm, the grammar gives me the following error

Reply:
The right-hand-side of 'join' should be a "projection_expression" not a "join_expression". `join` is left-associative in the grammar.